src/InstrumentationEngine.Lib/LoggerService.cpp

```diff
@@ -133,8 +133,13 @@ void CLoggerService::SetLogToDebugPort(_In_ bool enable)
     RecalculateLoggingFlags();
 }
 
-HRESULT CLoggerService::Initialize()
+HRESULT CLoggerService::Initialize(_In_ std::function<void(const LoggingFlags&)> callback)
 {
+    if (callback != nullptr)
```

Comment:
>if (callback != nullptr) [](start = 4, length = 24)

Don't do this check. It will currently cause m_LoggingFlagsCallback to be random memory if callback == nullptr.

Reply:
Actually, std::function might initialize to null. I'm not sure how that works, to be honest.


---
In reply to: [423354284](https://github.com/microsoft/CLRInstrumentationEngine/pull/264#discussion_r423354284) [](ancestors = 423354284)

Reply:
Still, get rid of the check :-). It isn't needed.

---
In reply to: [423355501](https://github.com/microsoft/CLRInstrumentationEngine/pull/264#discussion_r423355501) [](ancestors = 423355501,423354284)

Reply:
Thanks